clang/lib/AST/ByteCode/Interp.cpp

```diff
@@ -348,6 +348,12 @@ bool CheckConstant(InterpState &S, CodePtr OpPC, const Descriptor *Desc) {
   if (D->isConstexpr())
     return true;
 
+  if (const auto *VD = dyn_cast_if_present<VarDecl>(S.EvaluatingDecl);
```

Comment:
```suggestion
  // If we're evaluating the initializer for a constexpr variable in C23, we may only read other contexpr variables. Abort here since this one isn't constexpr.
  if (const auto *VD = dyn_cast_if_present<VarDecl>(S.EvaluatingDecl);
```